bookwyrm/views/hashtag.py

```diff
@@ -0,0 +1,54 @@
+""" listing statuses for a given hashtag """
+from django.core.paginator import Paginator
+from django.db.models import Q
+from django.views import View
+from django.shortcuts import get_object_or_404
+from django.template.response import TemplateResponse
+
+from bookwyrm import models
+from bookwyrm.settings import PAGE_LENGTH
+from bookwyrm.views.helpers import maybe_redirect_local_path
+
+
+# pylint: disable= no-self-use
+class Hashtag(View):
+    """listing statuses for a given hashtag"""
+
+    # pylint: disable=unused-argument
+    def get(self, request, hashtag_id, slug=None):
+        """show hashtag with related statuses"""
+        hashtag = get_object_or_404(models.Hashtag, id=hashtag_id)
+
+        if redirect_local_path := maybe_redirect_local_path(request, hashtag):
+            return redirect_local_path
+
+        activities = (
+            models.Status.privacy_filter(
+                request.user,
+            )
+            .filter(
+                Q(mention_hashtags=hashtag),
+            )
+            .exclude(
+                privacy__in=["direct", "unlisted"],
+            )
```

Comment:
The one gotcha with this that makes this view complicated is that it will end up showing public statuses by users who have blocked one another. It ends up getting really complicated! That's why the list of lists and the timelines use a redis cache system, which could be extended without too much trouble (at least for me because presumably I still understand that code) to this view. But that would probably be out of scope for this PR..

Reply:
Ah good point. I'll give it a try and see how much work it would be to handle blocked users correctly. Might take me a bit though as I'll be too busy next week

Reply:
There's some existing code for that which would at least be useful reference; I can dig it up tomorrow. Feel free to remind me if I don't!

Reply:
That would be great, but no hurry. I won't be able to spend any time on this all of next week.

Reply:
Here we go: https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/models/base_model.py#L155

Reply:
I think that's already applied, the view uses `models.Status.privacy_filter` which in turns calls `super().privacy_filter` ([here](https://github.com/chdorner/bookwyrm/blob/63f0af9c603cae7b30b6e35a608de2d8aa11e3fa/bookwyrm/models/status.py#L230)) from `BookWyrmModel` which is the one you linked, [this](https://github.com/chdorner/bookwyrm/blob/259e8ef8f6fd9bcd5580ba750d2c0755bba47964/bookwyrm/models/base_model.py#L142) one.

Did you see statuses from users who blocked each other show up in your local testing? I'll make sure to give it a test run locally before marking this PR as ready for review / merging.

Reply:
Ok, tested this out locally.

With the following users and blocking relationships:
* `admin` does not block anybody, is not blocked by anybody
* `auser` does not block anybody, is blocked by `buser`
* `buser` blocks `auser`, is not blocked by anybody

all three users have one status posted using the same hashtag. The visibility of the statuses on the hashtag page looks like this:
* `admin` sees all three statuses
* `auser` only sees their own status and the one from `admin`
* `buser` only sees their own status and the one from `admin`

Sounds like it's working correctly, is that right?

Reply:
That sounds correct!